seed/analysis_pipelines/pipeline.py

```diff
@@ -31,49 +31,166 @@ def task_create_analysis_property_views(analysis_id, property_view_ids, progress
+                    }
+                    logger.info(json.dumps(log_message))
+                    _stop_task_chain(_self)
+                    return
+                else:
+                    # something Bad has happened
+                    raise AnalysisPipelineException(f'When preparing to run the task {func}, expected analysis status to be {expected_status} but it was {analysis.status}')
+            except Analysis.DoesNotExist:
+                # someone deleted the analysis
+                log_message = {
+                    'analysis_id': _analysis_id,
+                    'debug_message': 'Analysis no longer exists before starting the task. '
+                                     'Assuming the analysis was deleted and stopping the celery task chain.'
+                }
+                logger.info(json.dumps(log_message))
+                _stop_task_chain(_self)
+                return
+
+            # Catch all exceptions raised by the task
+            # If the exception was to stop the task chain, stop the task chain
+            # If the analysis no longer exists, ignore the exception (the analysis was deleted by someone else)
+            # If the analysis _does_ still exist, raise the exception (something unexpected happened)
+            try:
+                return func(*args, **kwargs)
+            except StopAnalysisTaskChain as e:
+                log_message = {
+                    'analysis_id': _analysis_id,
+                    'debug_message': 'StopAnalysisTaskChain exception raised, stopping the celery task chain.',
+                    'exception': repr(e)
+                }
+                logger.info(json.dumps(log_message))
+                _stop_task_chain(_self)
+                return
+            except Exception as e:
+                try:
+                    Analysis.objects.get(id=_analysis_id)
+
+                    log_message = {
+                        'analysis_id': _analysis_id,
+                        'debug_message': 'Caught unexpected exception occurred during analysis task (and analysis still exists). Re-raising exception and continuing.',
+                        'exception': repr(e)
+                    }
+                    logger.error(json.dumps(log_message))
+                    raise e
+                except Analysis.DoesNotExist:
+                    # someone deleted the analysis, and the error was probably due to that
+                    log_message = {
+                        'analysis_id': _analysis_id,
+                        'debug_message': 'Task raised unhandled exception, but the analysis no longer exists. '
+                                         'Assuming the analysis was deleted and stopping the celery task chain.',
+                        'exception': repr(e)
+                    }
+                    logger.info(json.dumps(log_message))
+                    _stop_task_chain(_self)
+                    return
```

Comment:
I'm surprised to see this again as I would've thought the first existence check would be enough. Is this to accommodate for if the Analysis is deleted while the function was being run?

Reply:
Yes, this is to handle the case where the task is still running, someone else deleted the analysis, and as a result some unexpected exception is raised in the task (maybe we tried to link an analysis message or file to the analysis). The first existence check is to avoid starting a task when the analysis was already deleted.